### DISCUSSION TITLE
Dynamic Switch of local and remote rendering. (on object interaction).

### DISCUSSION THREAD

**Discussion by Jcdoliner** [Kitware/trame#394] [Trame/ParaView]
I have been working on a project with trame/Paraview for the past couple of months, but I'm concerned about the performance of the tool. Using Paraview as a backend, Local Rendering, as documentation suggests, is definitely faster than remote. The issue that I have is that my tool requires annotations and the color bar that are not available to me when I use Local rendering. 

I first had the idea to have the tool automatically switch to local rendering when the user interacts with the object therefore allowing fast performance and color bars after the user interacts. After several attempts I was unable to get it working. 

In a nutshell, I want my code to trigger a callback (bind a controller) when I interact with the rendering and one when I end the interaction. The controller will then take care of switching the rendering mode (which I can already do in my code). The click parameter does not work as well as the other events listed in the VtkRemoteLocalView Documentation. 

I would appreciate if anybody has an example or if anyone knows if this is possible. Thanks

**Comment by jourdain** [comment]
That is definitely possible to achieve but will require some fine-tuning and experiments. You may even have to manage your own Remote/LocalView yourself to better manage the switch.

Out of curiosity, why is the remote rendering slow? Do you have a GPU on the server side?

**Reply by Jcdoliner** [reply]
Yes there is a gpu on the servers side, I don't think that's the problem.
The app is running using Paraview from Conda, I don't know if that might cause the lack of performance.

**Reply by jourdain** [reply]
Remote rendering with GPU is pretty fast. 
Is the server a Linux? Do you know if conda build ParaView with EGL?
You can a downloaded version of ParaView with your conda if you set LD_LIBRARY_PATH + PYTHONPATH to point to the proper ParaView directories (assuming the Python version is compatible.
